bindings/rust/bench/src/harness.rs

```diff
@@ -12,14 +19,145 @@ pub enum Mode {
     Server,
 }
 
-pub trait TlsBenchHarness {
+// these parameters were the only ones readily usable for all three libaries:
+// s2n-tls, rustls, and openssl
+#[allow(non_camel_case_types)]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum CipherSuite {
+    AES_128_GCM_SHA256,
+    AES_256_GCM_SHA384,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ECGroup {
+    SECP256R1,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct CryptoConfig {
+    pub cipher_suite: CipherSuite,
+    pub ec_group: ECGroup,
+}
+
+pub trait TlsBenchHarness: Sized {
```

Comment:
Have you considered how you would expand the benchmarking test to run a TLS12 test?

Reply:
No, I think that might be outside the scope of my project.

Reply:
This is a test we would definitely want, since many people still use TLS12. Maybe it's not a part of your project plan, but it is important for your code to be flexible enough to work for other usecases. This is not really a critique of the current code, but it would be a good exercise to think about how this test could be expanded to TLS1.2. I assume it would be pretty easy, you'd just want to define some different cipher suites.

Reply:
I agree, it shouldn't be hard: version could be passed into `new()` as a parameter and we change the configs depending on the version. For now, I don't want to change the names of the cipher suites to reflect the version because everything currently (and in the next few weeks) works under the assumption we're using TLS1.3.